store/iavl/store.go

```diff
@@ -29,13 +31,19 @@ var (
 
 // Store Implements types.KVStore and CommitKVStore.
 type Store struct {
-	tree Tree
+	tree       Tree
+	pruning    types.PruningOptions
+	version    int64
+	lastCommit types.CommitID
 }
 
 // LoadStore returns an IAVL Store as a CommitKVStore. Internally it will load the
 // store's version (id) from the provided DB. An error is returned if the version
 // fails to load.
 func LoadStore(db dbm.DB, id types.CommitID, pruning types.PruningOptions, lazyLoading bool) (types.CommitKVStore, error) {
+	if !pruning.IsValid() {
+		panic(fmt.Sprintf("PruningOptions are invalid: %#v", pruning))
```

Comment:
Why do we panic here when the function signature returns an error?

```suggestion
		return nil, fmt.Errorf("pruning options are invalid: %#v", pruning))
```